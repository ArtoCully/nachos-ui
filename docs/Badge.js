import React from 'react'
import { View } from 'react-native'

import Badge from '../src/Badge'
import H4 from '../src/typography/H4'

// WEBPACK (BadgeExample)
const BadgeExample = () => {
  return (
    <View>
      <H4>Example:</H4>
      <View style={{ margin: 15, flexDirection: 'row' }}>
        <Badge value={123} style={{ marginRight: 15 }} />
        <Badge value={4} color='red' />
      </View>
    </View>
  )
}
// WEBPACK

const description = `
| Prop | Type | Default | Description |
|------|------|---------|-------------|
| \`color\` | \`string\` | \`null\` | Background color of the Badge compoment |
| \`value\` | \`string, number\` | \`void\` | The value that's shown inside the compoment |
| \`style\` | \`object, array\` | \`void\` | Style of the Badge compoment |
| \`textStyle\` | \`object, array\` | \`void\` | Style of the Text inside the Badge compoment |
| \`theme\` | \`object\` | \`defaultTheme\` | An object describing a style of the component. See more in Theme section.  |
`

BadgeExample.styleguide = {
  ...BadgeExample.styleguide,
  title: 'Badge',
  description,
}

export default BadgeExample
