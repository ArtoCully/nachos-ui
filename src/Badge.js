import React, { PropTypes } from 'react'
import { Text, View } from 'react-native'
import themeManager from './themeManager'

const defaultTheme = {
  BADGE_BACKGROUND: '#444',
  BADGE_BORDER_RADIUS: 20,
  BADGE_FONT_COLOR: '#fff',
  BADGE_FONT_SIZE: 14,
}

themeManager.setSource('Badge', () => defaultTheme)

const defaultStyle = (theme) => {
  return {
    container: {
      flexDirection: 'row',
    },
    base: {
      padding: 10,
      paddingTop: 3,
      paddingBottom: 3,
      backgroundColor: theme.BADGE_BACKGROUND,
      borderRadius: theme.BADGE_BORDER_RADIUS,
    },
    text: {
      fontSize: theme.BADGE_FONT_SIZE,
      color: theme.BADGE_FONT_COLOR,
    },
  }
}

const Badge = (props) => {
  const {
    color,
    style,
    textStyle,
    value,
  } = props

  const theme = props.theme || themeManager.getStyle('Badge')
  const baseStyle = defaultStyle(theme)

  return (
    <View style={baseStyle.container}>
      <View
        style={[
          baseStyle.base,
          color ? { backgroundColor: color } : {},
          style,
        ]}
      >
        <Text style={[baseStyle.text, textStyle]}>
          {value}
        </Text>
      </View>
    </View>
  )
}

Badge.propTypes = {
  color: PropTypes.string,
  style: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  textStyle: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]).isRequired,
  theme: PropTypes.object,
}

export default Badge
