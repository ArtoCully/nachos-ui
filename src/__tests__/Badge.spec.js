import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'

import Badge from '../Badge'

it('should render correctly', () => {
  const component = renderer.create(<Badge value={4} color='red' />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
