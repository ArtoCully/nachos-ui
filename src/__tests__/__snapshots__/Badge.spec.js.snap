exports[`test should render correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#444",
          "borderRadius": 20,
          "padding": 10,
          "paddingVertical": 4,
        },
        Object {
          "backgroundColor": "red",
        },
        undefined,
      ]
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontSize": 12,
          },
          undefined,
        ]
      }>
      4
    </Text>
  </View>
</View>
`;
